fdbserver/DataDistributionTracker.actor.cpp

```diff
@@ -822,7 +822,8 @@ ACTOR Future<Void> fetchShardMetricsList_impl( DataDistributionTracker* self, Ge
 			// list of metrics, regenerate on loop when full range unsuccessful
 			Standalone<VectorRef<DDMetricsRef>> result;
 			Future<Void> onChange;
-			for (auto t : self->shards.containedRanges(req.keys)) {
+			for (auto t = self->shards.containedRanges(req.keys).begin();
+			     t != self->shards.intersectingRanges(req.keys).end(); ++t) {
```

Comment:
Can we avoid doing calling `intersectingRanges` every loop iteration?

Reply:
Yeah, this makes sense. Done, thanks.